app/controllers/manageBusinesses/manage/CheckYourAnswersController.scala

```diff
@@ -0,0 +1,230 @@
+              val taxYearStart = taxYear - 1
+              val taxYearEnd = taxYear
+              val incomeSourceBusinessName: Option[String] = user.incomeSources.getIncomeSourceBusinessName(incomeSourceType, incomeSourceIdStringOpt)
+              val incomeSourceIdOpt = incomeSourceIdStringOpt.map(id => IncomeSourceId(id))
+              val incomeSourceId: Option[IncomeSourceId] = user.incomeSources.getIncomeSourceId(incomeSourceType, incomeSourceIdOpt.map(m => m.value))
+
+              handleSubmitRequest(errorCall, isAgent, successCall, taxYearStart, taxYearEnd, incomeSourceId, ReportingMethod(reportingMethod), incomeSourceBusinessName, incomeSourceType)
+            case _ => Future.successful(logAndShowError(isAgent, s"[handleSubmitRequest]: Missing session values"))
+          }
+        }
+      }
+    }
+  }
+
+  private def handleSubmitRequest(errorCall: Call,
+                                  isAgent: Boolean,
+                                  successCall: Call,
+                                  taxYearStart: Int,
+                                  taxYearEnd: Int,
+                                  incomeSourceIdMaybe: Option[IncomeSourceId],
+                                  reportingMethod: ReportingMethod,
+                                  incomeSourceBusinessName: Option[String],
+                                  incomeSourceType: IncomeSourceType
+                                 )(implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
+
+    val updateIncomeSourceResFuture = updateIncomeSource(taxYearEnd, incomeSourceIdMaybe, reportingMethod)
+
+    updateIncomeSourceResFuture flatMap {
+      case _: UpdateIncomeSourceResponseError =>
+        logAndShowError(isAgent, "Failed to update reporting method")
+        auditingService
+          .extendedAudit(
+            ManageIncomeSourceCheckYourAnswersAuditModel(
+              isSuccessful = false,
+              journeyType = incomeSourceType.journeyType,
+              operationType = "MANAGE",
+              reportingMethodChangeTo = formatReportingMethod(reportingMethod),
+              taxYear = taxYearStart.toString + "-" + taxYearEnd.toString,
+              businessName = incomeSourceBusinessName.getOrElse("Unknown")
+            )
+          )
+        Future.successful(Redirect(errorCall))
+      case _: UpdateIncomeSourceResponseModel =>
+        (incomeSourceIdMaybe) match {
+          case Some(incomeSourceId) =>
+            withSessionData(JourneyType(Manage, incomeSourceType), journeyState = BeforeSubmissionPage) {
+              uiJourneySessionData =>
+                val newUIJourneySessionData = {
+                  uiJourneySessionData.copy(manageIncomeSourceData =
+                    Some(ManageIncomeSourceData(Some(incomeSourceId.value), Some(reportingMethod.name), Some(taxYearEnd), Some(true))))
+                }
+                sessionService.setMongoData(newUIJourneySessionData)
```

Comment:
line 166: unhandled Future, this could cause problems in the future, we must handled each future response;